### PR TITLE
Feature/adq patch final

### PR DESCRIPTION
**Pull request by flomnes**
See #785, #903 and #961 for comments.

See #1083 for discussion about what's left to do.

### CODE REVIEW

File: docs/reference-guide/14-adequacy-patch.md [hunk 1]
```diff
@@ -53,8 +53,185 @@ Once the first iteration is completed, we have access to the DENS value, which r
...
+
+   - ENS (node A) $\ge$ 0 
+   - spillage (node A) $\ge$ 0 
+
+_**Notes**_
+
+- _“Spillage” variable and “Spillage_init” parameter have been introduced only to deal with some situations for which “Flowbased” constraints, combining with adequacy patch rules, lead to an increase of Total ENS over the different nodes “2”. Such increase of Total ENS could happen for 2 reasons:_ 
+
+   - _We have new violations of Local Matching rule and the optimal solution found by Antares is no longer a valid solution, regards to this rule;_ 
+   - _The curtailment sharing rule target is to minimize $\sum$(ENS$^2$/PTO) and such objective is not exactly equivalent than minimizing Total ENS._
```
Comment (guilpier-code) [inline docs/reference-guide/14-adequacy-patch.md:175]: `$\sum$(ENS$^2$/PTO)` does not render properly on my **VSCode**.
And it's also the case for many scientific formulas after this one.

File: src/libs/antares/study/parameters.cpp [hunk 2]
```diff
@@ -207,6 +206,39 @@ const char* StudyModeToCString(StudyMode mode)
     }
     return "Unknown";
 }
+bool StringToPriceTakingOrder(const AnyString& PTO_as_string, AdequacyPatch::AdqPatchPTO& PTO_as_enum)
```
Comment (guilpier-code) [inline src/libs/antares/study/parameters.cpp:209]: The parameter.cpp file is growing bigger and bigger.
What do you think of removing things to separate **.h/.cpp** files ?
And it's the same for **fwd.h**.
For instance, about **adq patch**, move free functions and **enum** definition to **adq-patch-params.h/.cpp**.

File: src/libs/antares/study/parameters.h [hunk 3]
```diff
@@ -136,6 +135,10 @@ class Parameters final
     */
     void resetSeeds();
     /*!
+    ** \brief Reset to default all threshold values in adequacy patch
+    */
+    void resetThresholdsAdqPatch();
```
Comment (guilpier-code) [inline src/libs/antares/study/parameters.h:140]: See remarks in **parameters.cpp**

**Review by guilpier-code — APPROVED** [review]

File: src/libs/antares/study/parameters.cpp (hunk 2 above)
Comment (flomnes) [reply src/libs/antares/study/parameters.cpp:209]: I agree

File: src/libs/antares/study/parameters.h (hunk 3 above)
Comment (flomnes) [reply src/libs/antares/study/parameters.h:140]: I agree

File: docs/reference-guide/14-adequacy-patch.md (hunk 1 above)
Comment (flomnes) [reply docs/reference-guide/14-adequacy-patch.md:175]: This is meant to be rendered in PDF or [online](https://antares-simulator.readthedocs.io/en/latest/reference-guide/14-adequacy-patch/)

**Pull request merged**